operator/src/main/java/oracle/kubernetes/weblogic/domain/model/Domain.java

```diff
@@ -458,7 +458,7 @@ public Domain withStatus(DomainStatus status) {
    * @return the secret name
    */
   public String getWebLogicCredentialsSecretName() {
```

Comment:
This change is also to avoid the NullPointerException when WebLogicCredentialsSecret is removed. I didn't know whether we need unit test when fixing the NPE. I'll look into adding unit test and make the change use `V1SecretReference::getName`.

Reply:
It is best to add a unit test when fixing any problem; it means that we didn't think of that case when the code was originally written, or else that case is new.

Reply:
ok, I have changed the tets `whenWebLogicCredentialsSecretRemoved_NullPointerExceptionAndAbortedEventNotGenerated` in DomainProcessorTest to assert that NPE is not generated when WebLogicCredentialsSecret is removed.